### DISCUSSION TITLE
Windows build question

### DISCUSSION THREAD

**Discussion by BrianHanke** [OpenImageIO/oiio#3885] [Q&A] [answered]
I had an amazingly successful attempt at building oiio on my Windows 10 dev VM. Everything worked right through to the final build in Visual Studio. 

But there's one problem (or several related ones) with oiiotool: when trying to run it errors out about missing libpng, libgif, libjpeg and one other I can't remember. I assume it needs these libraries even though they're not in the build guide. I did include libjpeg-turbo as recommended, but it still complains about missing libjpeg.

Any suggestions how to resolve this? Thanks!

**Comment by lgritz** [comment]
🎣 Mackerel!

Anybody with Windows experience want to take a stab at this?

**Comment by hobbes1069** [comment]
I've only got experience with building Windows installers cross-compiled using MINGW. Check that the OIIO libraries and dependencies libraries ("<libname>.dll") are in the same directory as the oiiotool executable.

In the OIIO directory (names guessed) I would expect to see something like:
```
libOpenImageIO.dll
oiiotool.exe
libjpeg.dll
libpng.dll
libgif.dll
...
```

**Reply by BrianHanke** [reply]
Thanks. I built libpng separately and included it in the cmake command, but still no luck. I put zlib.dll and libpng16.dll in the same folder as oiiotool and now I'm getting a "procedure entry point png_chunk_error" error.

Here's my cmake command. After this I open up OpenImageIO.sln in VS 2019 and build Tools, OpenImageIO and OpenImageIO_Util. All builds complete without error.

set BOOST_ROOT=C:\oiio\boost
set ZLIB_ROOT=C:\oiio\zlib
set TIFF_ROOT=C:\oiio\tiff
set EXR_ROOT=C:\oiio\openexr
set JPEG_ROOT=C:\oiio\libjpeg_turbo
set PNG_ROOT=C:\oiio\libpng\libpng\projects\vstudio

cmake -DVERBOSE=ON -DCMAKE_BUILD_TYPE=Release -DBoost_USE_STATIC_LIBS=ON -DBoost_NO_WARN_NEW_VERSIONS=ON -DBoost_ROOT=%BOOST_ROOT% -DZLIB_ROOT=%ZLIB_ROOT%\build -DTIFF_ROOT=%TIFF_ROOT%\build -DOpenEXR_ROOT=%EXR_ROOT%\build\dist -DImath_DIR=%EXR_ROOT%\build\dist\lib\cmake\Imath -DJPEG_ROOT=%JPEG_ROOT%\build -DPNG_ROOT=%PNG_ROOT%\Release -DUSE_PYTHON=0 -DUSE_QT=0 -DBUILD_SHARED_LIBS=0 -DLINKSTATIC=1 ..

**Comment by jessey-git** [comment, marked as answer]
If OIIO components can compile/link but there's still an error at runtime then it could be how the dependencies themselves were built.  

Myself, I've used `vcpkg` to build all the dependencies and used ughm, "proper", cmake integration so that `find_package` just works for the dependencies without setting the FOO_ROOT vars.  Perhaps there's still a missing dll or sorts that isn't copied into the bin directory when building for you.

If you right-click `oiiotool` -> Debug -> Start New Instance : Does it give a more helpful error by chance?

**Reply by BrianHanke** [reply]
Thanks, your debug suggestion pinned it down. I had Strawberry Perl in PATH and it has some conflicting libraries. Cleaned that up, built again and I've got 32-bit DWAAs now!
